client/latis.py

```diff
@@ -61,31 +75,46 @@ class Dataset:
     def __init__(self, latisInstance, name):
         self.latisInstance = latisInstance
         self.name = name
+        self.selection = []
+        self.projection = []
+        self.operation = []
         self.query = None
```

Comment:
I would just use the `buildQuery` function as needed (e.g. when reading data) instead of holding onto it here.